src/catalog/postgres/builder.cpp

```diff
@@ -548,4 +589,136 @@ IndexSchema Builder::GetConstraintForeignTableIndexSchema(db_oid_t db) {
+
+  columns.emplace_back("pronargs", type::TypeId::SMALLINT, false, MakeNull(type::TypeId::TINYINT));
+  columns.back().SetOid(PRONARGS_COL_OID);
+
+  columns.emplace_back("pronargdefaults", type::TypeId::SMALLINT, false, MakeNull(type::TypeId::TINYINT));
+  columns.back().SetOid(PRONARGDEFAULTS_COL_OID);
+
+  columns.emplace_back("prorettype", type::TypeId::INTEGER, false, MakeNull(type::TypeId::INTEGER));
+  columns.back().SetOid(PRORETTYPE_COL_OID);
+
+  columns.emplace_back("proargtypes", type::TypeId::INTEGER, false, MakeNull(type::TypeId::INTEGER));
+  columns.back().SetOid(PROARGTYPES_COL_OID);
+
+  columns.emplace_back("proallargtypes", type::TypeId::VARBINARY, 4096, false, MakeNull(type::TypeId::VARBINARY));
+  columns.back().SetOid(PROALLARGTYPES_COL_OID);
+
+  columns.emplace_back("proargmodes", type::TypeId::VARBINARY, 4096, false, MakeNull(type::TypeId::VARBINARY));
+  columns.back().SetOid(PROARGMODES_COL_OID);
+
+  columns.emplace_back("proargdefaults", type::TypeId::VARBINARY, 4096, false, MakeNull(type::TypeId::VARCHAR));
+  columns.back().SetOid(PROARGDEFAULTS_COL_OID);
+
+  columns.emplace_back("proargnames", type::TypeId::VARBINARY, 4096, false, MakeNull(type::TypeId::VARBINARY));
+  columns.back().SetOid(PROARGNAMES_COL_OID);
+
+  columns.emplace_back("prosrc", type::TypeId::VARCHAR, 4096, false, MakeNull(type::TypeId::VARCHAR));
+  columns.back().SetOid(PROSRC_COL_OID);
+
+  columns.emplace_back("proconfig", type::TypeId::VARBINARY, 4096, false, MakeNull(type::TypeId::VARBINARY));
+  columns.back().SetOid(PROCONFIG_COL_OID);
+
+  return Schema(columns);
+}
+
+IndexSchema Builder::GetProcOidIndexSchema(db_oid_t db) {
+  std::vector<IndexSchema::Column> columns;
+
+  columns.emplace_back("prooid", type::TypeId::INTEGER, false,
+                       parser::ColumnValueExpression(db, PRO_TABLE_OID, PROOID_COL_OID));
+  columns.back().SetOid(indexkeycol_oid_t(1));
+
+  // Primary
+  IndexSchema schema(columns, storage::index::IndexType::HASHMAP, true, true, false, true);
+
+  return schema;
+}
+
+IndexSchema Builder::GetProcNameIndexSchema(db_oid_t db) {
+  std::vector<IndexSchema::Column> columns;
+
+  columns.emplace_back("pronamespace", type::TypeId::INTEGER, false,
+                       parser::ColumnValueExpression(db, PRO_TABLE_OID, PRONAMESPACE_COL_OID));
+  columns.back().SetOid(indexkeycol_oid_t(1));
+
+  columns.emplace_back("proname", type::TypeId::VARCHAR, MAX_NAME_LENGTH, false,
+                       parser::ColumnValueExpression(db, PRO_TABLE_OID, PRONAME_COL_OID));
+  columns.back().SetOid(indexkeycol_oid_t(2));
+
+  columns.emplace_back("proallargs", type::TypeId::VARBINARY, MAX_NAME_LENGTH, false,
+                       parser::ColumnValueExpression(db, PRO_TABLE_OID, PROALLARGTYPES_COL_OID));
+  columns.back().SetOid(indexkeycol_oid_t(3));
+
+  // Unique, not primary
+  IndexSchema schema(columns, storage::index::IndexType::HASHMAP, true, false, false, true);
```

Comment:
This should be the default bwtree index because we may need to do range scans on the index (e.g. all procs in a given namespace).